lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -115,17 +122,33 @@ void pthread_scheduler_wait_cq (cl_command_queue cq)
           return;
         }
       POCL_UNLOCK_OBJ (cq);
-      pthread_cond_wait (&scheduler.cq_finished_cond,
-                         &scheduler.cq_finished_lock);
-      pthread_mutex_unlock (&scheduler.cq_finished_lock);
+
+#ifdef HAVE_CLOCK_GETTIME
```

Comment:
I'm not sure if it's a race or something else, i spent some time on this but i'm unfamiliar with the pthread driver code, so dunno yet. Without this patch though, the integer_ops conformance tests always get stuck (for some reason they seem much more threaded than other tests).